### PR TITLE
add estimate total rows count to metrics for dump table

### PR DESCRIPTION
**Pull request by recall704**
### What problem does this PR solve? 
add estimate total rows count

### What is changed and how it works?
add prometheus metrics to save estimate_total_rows before you dump data

### Limitation
it does not work for `-S or --sql` and views data

### CODE REVIEW

File: v4/export/dump.go [hunk 1]
```diff
@@ -209,6 +209,12 @@ func (d *Dumper) Dump() (dumpErr error) {
 		}
 	})
 
+	// get estimate total count
+	err = d.getEstimateTotalRowsCount(tctx, metaConn)
+	if err != nil {
+		return err
```
Comment (lichunzhu) [inline v4/export/dump.go:215]: I think this isn't a fatal error. We can just `log.Error` here.

**Review by lichunzhu — COMMENTED** [review]
rest LGTM

File: v4/export/dump.go (hunk 1 above)
Comment (recall704) [reply v4/export/dump.go]: fixed

**Comment by lichunzhu** [comment]
Could you please help sign the CLA? https://github.com/pingcap/dumpling/pull/254#issuecomment-806522513

Thanks a lot for your contribution!

**Pull request merged**